agentos/repo.py

```diff
@@ -62,6 +62,21 @@ def from_spec(spec: NestedRepoSpec, base_dir: Path = None) -> "Repo":
             f"Unknown repo spec type '{repo_type} in repo {identifier}"
         )
 
+    @classmethod
+    def from_registry(
+        cls, registry: Registry, identifier: RepoIdentifier
+    ) -> "Repo":
+        return cls.from_spec(registry.get_repo_spec(identifier))
+
+    @classmethod
+    def from_github(
+        cls, github_account: str, repo_name: str, identifier: str = None
+    ) -> "GitHubRepo":
+        if not identifier:
+            identifier = f"{github_account}/{repo_name}"
+        url = f"https://github.com/{identifier}"
```

Comment:
If I pass `identifier` here, won't this result in a broken github url?  E.g. I think the call:

```python
 Repo.from_github('agentos-project', 'agentos', identifier='foo')
```

result in a broken repo object.

Reply:
Fixed!